sktime/transformations/series/holiday.py

```diff
@@ -0,0 +1,310 @@
+    categories : Tuple[str], optional
+        requested holiday categories.
+
+    References
+    ----------
+    .. [1] https://github.com/dr-prodigy/python-holidays
+
+    Examples
+    --------
+    >>> from sktime.transformations.series.holiday import CountryHolidaysTransformer
+    >>>
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>>
+    >>> y_t = CountryHolidaysTransformer("US").fit_transform(y)
+    >>> y_t.columns
+    Index(['Number of airline passengers', 'country_holidays'], dtype='object')
+    >>> y_t.dtypes.to_dict()
+    {'Number of airline passengers': dtype('float64'), 'country_holidays': dtype('bool')}  # noqa: E501
+    >>> y_t["country_holidays"].sum()
+    14
+    """
+
+    _tags = {
+        "scitype:transform-input": "Series",
+        "scitype:transform-output": "Series",
+        "scitype:instancewise": True,
+        "scitype:transform-labels": "None",
+        "X_inner_mtype": "pd.DataFrame",
+        "y_inner_mtype": "None",
+        "univariate-only": False,
+        "fit_is_empty": True,
+        "capability:inverse_transform": False,
+        "capability:unequal_length": True,
+        "handles-missing-data": False,
+        "python_version": ">=3.8",
+        "python_dependencies": ["holidays"],
+    }
+
+    def __init__(
+        self,
+        country,
+        subdiv=None,
+        years=None,
+        expand=True,
+        observed=True,
+        language=None,
+        categories=None,
+    ):
+        self.country = country
+        self.subdiv = subdiv
+        self.years = years
+        self.expand = expand
+        self.observed = observed
+        self.language = language
+        self.categories = categories
+
+        super().__init__()
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing the core logic, called from transform
+
+        Parameters
+        ----------
+        X : Series or Panel of mtype X_inner_mtype
+            if X_inner_mtype is list, _transform must support all types in it
+            Data to be transformed
+        y : Series or Panel, default=None
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        transformed version of X
+        """
+        from holidays import country_holidays
+
+        holidays_data = country_holidays(
+            self.country,
+            subdiv=self.subdiv,
+            years=self.years,
+            expand=self.expand,
+            observed=self.observed,
+            language=self.language,
+            categories=self.categories,
+        )
+
+        Xt = X.copy(deep=True)
+
+        if isinstance(Xt.index, pandas.DatetimeIndex):
+            dates = Xt.index
+        elif isinstance(Xt.index, pandas.PeriodIndex):
+            dates = Xt.index.to_timestamp()
+        else:
+            raise TypeError(f"index type is unsupported: {type(Xt.index)}")
```

Comment:
this might upset the tests...

* if index of `X` is integer, one might just assume "days since start of 1970" (unix convention)
* shouhld the index of `Xt` not be the same as that of `X` (unless `expand=True`), or am I missing sth?

Reply:
May be I'm not following you, `Xt` does have same index as that of `X`. The conversion is only for `dates` which is local to this method, and not stored in either `X` or `Xt`.

---

Not sure about first point either. If indices are integers, are you suggesting to assume that they mean date difference since unix epoch? I'm not sure if that'd be a safe assumption, and I'd prefer to let it fail preferring explicit over implicit.

Reply:
I took a look at `DateTimeFeatures`, which looked similar.

https://github.com/sktime/sktime/blob/b29e147b54959a53cc96e5be9c3f819717aa38e7/sktime/transformations/series/date.py#L186-L192

Reply:
> May be I'm not following you, `Xt` does have same index as that of `X`. The conversion is only for dates which is local to this method, and not stored in either `X` or `Xt`.

My bad - I thought I saw you writing to the `index`, but I was mistaken.

Still, the error is raised for any non-datelike index, that much is right, no?

Reply:
> Not sure about first point either. If indices are integers, are you suggesting to assume that they mean date difference since unix epoch? I'm not sure if that'd be a safe assumption, and I'd prefer to let it fail preferring explicit over implicit.

Well, if the tests get upset about this. I thought they would?

Minor comment: in an exception like this: it is always good to say which estimator the `ValueError` is coming from; what the "found" an "expected" is.